crates/ln-dlc-node/src/ln/event_handler.rs

```diff
@@ -175,22 +174,38 @@ impl EventHandler {
                 fee_paid_msat,
                 ..
             } => {
```

Comment:
@luckysori: It appears that the outbound payment is never added to the the `self.outbound_payments`(At least I did not find any code for it). Hence the tests were not able to assert on a failed payment. I've adapted the code the code analogue to how we handle the Claimed Payment and insert the payment if `Vacant`. 

I suspected that this will also fix #328, but after a quick manual tests, it appears the issue still persists. Might be still something missing there. However, the tests are able to find the `outbound_payments` now.

Reply:
Good catch! I think we only ever tested manually with inbound payments 😨